src/ssh_cert_service/utils/ssh_keygen.py

```diff
@@ -0,0 +1,211 @@
+import os
+import tempfile
+import subprocess
+import re
+from pathlib import Path
+from typing import Any, Dict
+
+
+class SSHKeygen:
+    SSH_NAME = 'python_key'
+
+    def __init__(self, comment) -> None:
+        """ Instance variables
+
+            comment : string
+                Comment for the private key to be generated
+        """
+        self.comment = comment 
+
+    def gen_key(self,
+        passphrase: str = '',
+        identity: str = '',
+        domain: str = '',
+        validity: str = '',
+        principals: str = ''
+        ) -> tuple:
+        """ Generate a SSH Key
+            Parameters
+            ----------
+            comment : string
+                Comment for the private key to be generated
+            passphrase : string
+                password that will be added to the public key
+            indentity : string
+                String the helps to identified the signed key
+            domain : string
+                String that will be allowed the signed key 
+            validity : string
+                Expiration time for the signed key format -1d:+3w
+            principals : string
+                The user principals that will be allowed the signed key
+            Returns
+            ----------
+            tuple
+        """
+
+        # Create temporary dicrectory and storage the keys there
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            keys_path = f'{tmp_dir}/{self.SSH_NAME}'
+            # Genarate private and public key
+            subprocess.run(f'ssh-keygen -t rsa -C "{self.comment}" -N "{passphrase}" -f {keys_path}', shell=True, capture_output=True)
```

Comment:
I will create it as a new task because this is a bit complicated due to the user needs to provide the passphrase for verifying the ssh-cert signature.